### ISSUE TITLE
`IO` Support

### ISSUE THREAD

**Issue by Jacob-Chmura** [shenyangHuang/openDG#16]
To start, let's just support CSV streams

**Comment by Jacob-Chmura** [comment]
> As a first step, I think supporting loading from csv file is enough, What changes do we need to support CSV streams? (or more exactly how is it different from loading csv file)

I agree supporting CSV is sufficient for now. To clarify, what I meant is that currently, the assumption is that `DGStorage` and its concrete implementations deal with event streams directly. To avoid convoluting the storage API (and since it is ultimately a private class / not client facing), I intended to create an `IO` interface that handles the reading/writing logic from disc to event stream format that could directly be used as a public library API.

To summarize, this IO API would be client-facing, and handle all the IO logic. Since we agree that csv support is a good start, this boils down to having a `read` and `write` method that converts csv to the event stream and vice versa.
